### DISCUSSION TITLE
calling TRex from MATLAB

### DISCUSSION THREAD

**Discussion by schlurch1** [mooch443/trex#193] [Questions]
Hello mooch (&everybody else), I am really enjoying TRex, especially for being so fast. Thereby i would like to integrate in my analysis and call it from MATLAB. This generally works with using the system function:
system(['call', 'C:\Python\Anaconda3', '\Scripts\activate.bat tracking & ', 'C:\Python\Anaconda3\envs\tracking\bin', '\trex.exe'])

but i cannot manage to set parameters, which is usually done with the '-' (i.e. sth like: -i <videosource> -meta_real_width ....) any idea which command has to be used?

MATLAB itself states: "You can specify a file to be opened using ./trex -i <filename>" - but this did not work

Best Regards

**Comment by schlurch1** [comment]
never mind, i made it! fyi:

system(['call ', 'C:\Python\Anaconda3', '\Scripts\activate.bat tracking & ', ...
    'C:\Python\Anaconda3\envs\tracking\bin', '\trex.exe ', ...
    '-i ', input2, ...
    '-meta_real_width ', meta_real_width, ...
    '-blob_size_ranges ', blob_size_ranges, ...
    '-track_max_individuals ', track_max_individuals, ...
    '-track_threshold ', track_threshold, ...
    '-nowindow ', nowindow, ...
    '-output_dir', output, ...
    'auto_quit', autoquit, ...
    '&'])

Do not forget to place a space at the end of every parameter call and also at the end of the parameter itself

**Reply by mooch443** [reply]
Yes, this is more or less what I would have suggested. The syntax used in TRex of `-parameter value` is a very typical Unix/Linux syntax for passing arguments, where Windows would probably use `/parameter value` or something like that.

And I suppose if those spaces are really necessary there, it's because the only thing that `system`, given an array of strings, does, is probably to `"".join(array)`. If that is true though, you may be missing a couple. E.g. -output_dir and missing a dash at auto_quit. :-)
